src/runtime/process-worker.ts

```diff
@@ -20,12 +20,20 @@ export default class ProcessWorker implements TinypoolWorker {
   waitForExit!: Promise<void>
 
   initialize(options: Parameters<TinypoolWorker['initialize']>[0]) {
+    const workerId = options.workerData[0].workerId
+
     this.process = fork(
       fileURLToPath(import.meta.url + '/../entry/process.js'),
       options.argv,
-      options
+      {
+        ...options,
+        env: {
+          ...options.env,
+          TINYPOOL_WORKER_ID: workerId.toString(),
+        },
+      }
     )
-    this.threadId = this.process.pid!
+    this.threadId = workerId!
```

Comment:
This line should not change. When using `runtime: 'worker_threads'`, we set `Worker.threadId` here, which is not the same as Tinypool's internal `workerId`:
https://github.com/tinylibs/tinypool/blob/ac2e83917557df2deff74b9b6ea0c83319f32a45/src/runtime/thread-worker.ts#L16

```suggestion
    this.threadId = this.process.pid!
```

Reply:
Oh I misunderstood, so the `process.__tinypool_state__.workerId` is actually pulled out directly to get the "workerId"?

Reply:
Should be fixed now, I understand better now - vitest will grab the worker ID directly from the `process.__tinypool_state__.workedId` via the `workerId` export. Also added extra test to specifically test this which hopefully isn't too specific/brittle.